services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,108 @@
+package expandstorage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "POST":
+		handlePost(w, r)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(w http.ResponseWriter, r *http.Request) {
+	// When ContentLength is 0 that means request body is empty and
+	var err error
+	if r.ContentLength == 0 {
+		return
+	}
+
+	type poolDetails struct {
+		VolumeType           string `json:"volumeType"`
+		PoolName             string `json:"poolName"`
+		DataProtectionPolicy string `json:"dataProtectionPolicy"`
+		EnableCompression    bool   `json:"enableCompression"`
+	}
+
+	type storageClassDetails struct {
+		ReclaimPolicy     string `json:"reclaimPolicy"`
+		Name              string `json:"name"`
+		VolumeBindingMode string `json:"volumeBindingMode"`
+	}
+
+	var ExpandStorage = struct {
+		LsoStorageClassName string              `json:"lsoStorageClassName"`
+		EnableEncryption    bool                `json:"enableEncryption"`
+		PoolDetails         poolDetails         `json:"poolDetails"`
+		StorageClassDetails storageClassDetails `json:"storageClassDetails"`
+	}{}
+
+	if err = json.NewDecoder(r.Body).Decode(&ExpandStorage); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	client, err := client.New(ctrl.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		klog.Errorf("failed to create client: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	mutate := func() error { return nil }
+
+	// update storageCluster
+
+	// create cephBlockPool
+	createCephBlockPool(w, client, ExpandStorage.PoolDetails.PoolName, ExpandStorage.LsoStorageClassName, mutate)
+
+	// create storageClass
+}
+
+func createCephBlockPool(w http.ResponseWriter, client client.Client, poolName, lsoStorageClassName string, mutate func() error) {
+	cephBlockPool := &cephv1.CephBlockPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      poolName,
+			Namespace: "openshift-storage",
+		},
+		Spec: cephv1.NamedBlockPoolSpec{
+			PoolSpec: cephv1.PoolSpec{
+				FailureDomain: "host",
+				DeviceClass:   lsoStorageClassName,
+				Replicated: cephv1.ReplicatedSpec{
+					Size:                   3,
+					RequireSafeReplicaSize: true,
+				},
+			},
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(context.TODO(), client, cephBlockPool, mutate)
```

Comment:
you have missed adding the rbac as well you can add it here: https://github.com/red-hat-storage/ocs-operator/tree/main/config/rbac